### PR TITLE
Decrease the number of allocations for indexed reads

### PR DESCRIPTION
**Pull request by bheni**

### CODE REVIEW

**Comment by bheni** [comment]
#benchmark

**Comment by bheni** [comment]
#benchmark

**Comment by bheni** [comment]
#benchark

File: go/libraries/doltcore/table/map_point_reader.go [hunk 1]
```diff
@@ -53,3 +53,8 @@ func (pr *PointReader) Next(ctx context.Context) (k, v types.Value, err error) {
 
 	return k, v, err
 }
+
+// Next implements types.MapIterator.
```
Comment (bheni) [inline go/libraries/doltcore/table/map_point_reader.go:57]:
```suggestion
// NextTuple implements types.MapIterator.
```

**Comment by bheni** [comment]
#benchmark

File: go/store/types/sequence_cursor.go [hunk 2]
```diff
@@ -101,6 +102,12 @@ func (cur *sequenceCursor) current() (sequenceItem, error) {
 	return cur.getItem(cur.idx)
 }
 
+// current returns the value at the current cursor position
```
Comment (reltuk) [inline go/store/types/sequence_cursor.go:105]:
```suggestion
// currentTuple returns the tupleMapEntry at the current cursor position
```

File: go/libraries/doltcore/table/map_point_reader.go [hunk 3]
```diff
@@ -53,3 +53,8 @@ func (pr *PointReader) Next(ctx context.Context) (k, v types.Value, err error) {
 
 	return k, v, err
 }
+
+// NextTuple implements types.MapIterator.
+func (pr *PointReader) NextTuple(ctx context.Context) (k, v types.Tuple, err error) {
+	panic("not implemented")
```
Comment (reltuk) [inline go/libraries/doltcore/table/map_point_reader.go:59]: Should this return an error instead? Panics mean the code is busted...but maybe they're dangerous enough / likely enough in this case?

**Review by reltuk — APPROVED** [review]
LGTM! Couple nit comments.

**Comment by coffeegoddd** [comment]
#benchmark

**Pull request merged**